src/Akka.Hosting/AkkaConfigurationBuilder.cs

```diff
@@ -53,6 +52,8 @@ public enum HoconAddMode
     /// Delegate used to instantiate <see cref="IActorRef"/>s once the <see cref="ActorSystem"/> has booted.
     /// </summary>
     public delegate Task ActorStarter(ActorSystem system, IActorRegistry registry);
+    
+    public delegate Task ActorStarterWithResolver(ActorSystem system, IActorRegistry registry, IDependencyResolver resolver);
```

Comment:
new delegate type, since I didn't want to break backwards compat with everyone using the original `ActorStarter` (even though we're working on maturing the API). I didn't see any harm in adding a new delegate type here.